lib/src/timing.dart

```diff
@@ -0,0 +1,357 @@
+        (slices.last as SyncTimeTracker).isTracking;
+    var isExcludedNestedTrack =
+        !trackNested && zone[AsyncTimeTrackerImpl] != this;
+
+    // Exclude nested sync tracks
+    if (isNestedRun && isExcludedNestedTrack) {
+      var timer = slices.last as SyncTimeTrackerImpl;
+      slices.last = parent.run(zone, timer.splitNow);
+      try {
+        return action();
+      } finally {
+        parent.run(zone, timer.splitNow); // discard
+        slices.add(timer);
+      }
+    }
+
+    // Exclude nested async tracks
+    if (isExcludedNestedTrack) {
+      return action();
+    }
+
+    // Split time slices in nested sync runs
+    if (isNestedRun) {
+      return action();
+    }
+
+    var timer = SyncTimeTracker();
+    slices.add(timer);
+
+    // Pass to parent zone, in case of overwritten clock
+    return parent.runUnary(zone, timer.track, action);
+  }
+
+  static final asyncTimeTrackerZoneSpecification = ZoneSpecification(
+    run: <R>(Zone self, ZoneDelegate parent, Zone zone, R Function() f) {
+      var tracker = self[AsyncTimeTrackerImpl] as AsyncTimeTrackerImpl;
+      return tracker._trackSyncSlice(parent, zone, () => parent.run(zone, f));
+    },
+    runUnary: <R, T>(Zone self, ZoneDelegate parent, Zone zone, R Function(T) f,
+        T arg) {
+      var tracker = self[AsyncTimeTrackerImpl] as AsyncTimeTrackerImpl;
+      return tracker._trackSyncSlice(
+          parent, zone, () => parent.runUnary(zone, f, arg));
+    },
+    runBinary: <R, T1, T2>(Zone self, ZoneDelegate parent, Zone zone,
+        R Function(T1, T2) f, T1 arg1, T2 arg2) {
+      var tracker = self[AsyncTimeTrackerImpl] as AsyncTimeTrackerImpl;
+      return tracker._trackSyncSlice(
+          parent, zone, () => parent.runBinary(zone, f, arg1, arg2));
+    },
+  );
+
+  @override
+  R track<T, R extends FutureOr<T>>(R Function() action) {
+    if (isStarted) {
+      throw StateError('Can not be tracked twice');
+    }
+    _tracking = true;
+    var result = runZoned(action,
+        zoneSpecification: asyncTimeTrackerZoneSpecification,
+        zoneValues: {AsyncTimeTrackerImpl: this});
+    if (result is Future<T>) {
+      return result
+          // Break possible sync processing of future completion, so slice trackers can be finished
+          .whenComplete(() => Future.value())
+          .whenComplete(() => _tracking = false) as R;
+    } else {
+      _tracking = false;
+      return result;
+    }
+  }
+
+  bool _tracking;
+
+  @override
+  bool get isStarted => _tracking != null;
+
+  @override
+  bool get isFinished => _tracking == false;
+
+  @override
+  bool get isTracking => _tracking == true;
+}
+
+/// No-op implementation of [AsyncTimeTracker] that does nothing.
+///
+/// Use [AsyncTimeTracker.noOp] factory to get an instance.
+class NoOpAsyncTimeTracker implements AsyncTimeTracker {
+  static final sharedInstance = NoOpAsyncTimeTracker();
+
+  @override
+  Duration get duration => throw UnimplementedError();
```

Comment:
`UnimplementedError` -> `UnsupportedError`? It's not like we forgot – or we're goin to later. It's explicitly not supported, right?

Reply:
here and below...